### PR TITLE
[api][SIMS-#518] Open Award Eligibility in Camunda (CSG- FT)  -> Add variables for camunda

### PR DESCRIPTION
**Pull request by ann-aot**
- [x] Add variables to below

(from Offering) Offering_Intensity (PT/FT)
(from program) credential_type (degree, etc)
(from program) program_length (12WeeksToLessThan1Year, etc)
(from institution_location) instiuttion_location_type (Bc Private, etc)
(from institution_location) iinstitution_location_Province (BC,AB, etc)
(from student) student PD status (Y/N)
- [x] updated InstitutionType as an relation Object in Application
- [x] refactored the generic external API endpoint

### CODE REVIEW

File: sources/packages/api/src/database/entities/institution.model.ts [hunk 1]
```diff
@@ -78,9 +87,13 @@ export class Institution extends RecordDataModel {
   })
   users: InstitutionUser[];
 
-  @Column({
+  @RelationId((institution: Institution) => institution.institutionType)
+  institutionTypeId: number;
+
+  @ManyToOne(() => InstitutionType, { eager: false, cascade: true })
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/database/entities/institution.model.ts:93]: For this particular case, I believe that we should set the cascade to false.
The Institution Types are a fixed list that will not be changed normally and, if they changed, should not be in this way.

**Review by abschwenker — CHANGES_REQUESTED** [review]
As discussed, lets change to 1 call for get application with required details. This way camunda is consistent and we can re-use

File: sources/packages/api/src/route-controllers/application/application.system.controller.ts [hunk 2]
```diff
@@ -55,6 +54,38 @@ export class ApplicationSystemController {
...
+        mandatoryFees: application.offering.mandatoryFees,
+        exceptionalExpenses: application.offering.exceptionalExpenses,
+        tuitionRemittanceRequestedAmount:
+          application.offering.tuitionRemittanceRequestedAmount,
+        offeringDelivered: application.offering.offeringDelivered,
+        offeringIntensity: application.offering.offeringIntensity,
+      },
+      program: {
+        /**
+         * for now - if credential type is `other`, then the other is send to camunda
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/application/application.system.controller.ts:74]: I was not able to follow the comment here. Can you elaborate more on it?

File: sources/packages/api/src/route-controllers/application/models/application.model.ts [hunk 3]
```diff
@@ -39,6 +72,31 @@ export interface ApplicationDataDto {
    * Program Year is added to integrate the application in camunda workflows
    */
   programYear: string;
+
+  /**
+   * Offering details fro student application
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/application/models/application.model.ts:77]: Please fix the typo "fro" on this and other comments.
Add a "." period to the sentence.

File: sources/packages/api/src/services/application/application.service.ts [hunk 4]
```diff
@@ -324,16 +324,21 @@ export class ApplicationService extends RecordDataModelService<Application> {
     return application;
   }
 
+  /**
+   * get student application details with the application Id.
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/services/application/application.service.ts:328]: Please start the sentence with a upper case letter.

**Review by andrewsignori-aot — COMMENTED** [review]
Overall looks good 👍 
Please just reply to some minor comments.

File: sources/packages/api/src/services/application/application.service.ts [hunk 5]
```diff
@@ -324,16 +324,21 @@ export class ApplicationService extends RecordDataModelService<Application> {
...
-    const application = this.repo
-      .createQueryBuilder("application")
-      .select(["application", "programYear.programYear"])
-      .innerJoin("application.programYear", "programYear")
-      .where("programYear.active = true")
-      .andWhere("application.id = :applicationId", {
-        applicationId,
-      });
-    return application.getOne();
+    return this.repo.findOne(applicationId, {
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/services/application/application.service.ts:333]: Can we be more specific in the select clause here? It seems to me that we are retrieving way more than we need.

File: sources/packages/api/src/route-controllers/application/application.system.controller.ts (hunk 2 above)
Comment (ann-aot) [reply sources/packages/api/src/route-controllers/application/application.system.controller.ts:74]: In the program, when `credential_type` is select as `other`, users have the option to enter a free text credential_type. 
image
as of now, we are not sending the free text `credential_type` in the response. In the future, if there is a required, will need to update accordingly

**Review by abschwenker — COMMENTED** [review]
Please review tests, other than that lgtm

File: sources/packages/api/src/route-controllers/application/application.system.controller.ts (hunk 2 above)
Comment (abschwenker) [reply sources/packages/api/src/route-controllers/application/application.system.controller.ts:74]: perhaps say something like:

if credential type is `other`, then the string 'other' is sent to camunda. we are not sending the free text credential type when 'other' is selected

Comment (dheepak-aot) [reply sources/packages/api/src/route-controllers/application/application.system.controller.ts:74]: IMO template literals(`other`) can be avoided in the comment.

Comment (andrewsignori-aot) [reply sources/packages/api/src/route-controllers/application/application.system.controller.ts:74]: @ann-aot, I do not see a reason to have this comment at all.
I would recommend removing it.

**Review by abschwenker — APPROVED** [review]

**Review by dheepak-aot — APPROVED** [review]
LGTM 👍

**Review by andrewsignori-aot — APPROVED** [review]
Thanks for doing the changes. Looks good 👍

**Pull request merged**